### PR TITLE
[chore] Tiling pkg - improve test coverage

### PR DESCRIPTION
**Pull request by mozabes**
## Summary

Fills coverage gaps across `app/packages/tiling` identified in the post-sprint review. 16 new tests across 5 existing test files.

**Coverage before → after** (excluding stories):

| File | Before | After |
|------|--------|-------|
| `TilingProvider.tsx` | 93% | 100% |
| `use-tile-state.ts` | 71% | 100% |
| `TilingHeader.tsx` | 92% | 100% |
| `TilingInspectorSidebar.tsx` | 94% | 100% |
| **Overall** | **83%** | **88%** |

**What was added:**

- **`TilingProvider.test.tsx`** — explicit `initialLayout` prop; `setTileTitle` (update + no-op when unchanged); `removeTile` collapsing a split when the second of two tiles is removed; `removeTile` on a 3-tile layout where the sibling subtree survives; `useTiling()` error throw outside provider.
- **`use-tile-state.test.tsx`** — `useTileTitle` (scoped and unscoped); `useTileTitleFor` (explicit id and null id); `useSetTileTitle` (updates title, no-op outside scope).
- **`MosaicGrid.test.tsx`** — `addTileToLayout` throws when the new id already exists in the layout.
- **`TilingHeader.test.tsx`** — clicking a dropdown menu item calls `addTile` and adds the tile to the provider.
- **`TilingInspectorSidebar.test.tsx`** — Annotate tab renders its content; `formatSelection` falls back to `String()` for non-JSON-serialisable values (circular reference).

**Still at 0%**: `src/index.ts` (barrel export, not meaningful to test). `MosaicGrid.tsx` component code (61%) — the `TileWindow` component and drag/expand/close handlers depend on `react-mosaic-component`'s DnD backend which doesn't function in jsdom; left for integration tests.

## Test plan

- [ ] `cd app/packages/tiling && npx vitest run` — all 71 tests pass

## Summary by CodeRabbit

* **Tests**
  * Expanded test coverage for tile layout provider with validation of layout behavior, tile title updates, and tile removal scenarios.
  * Added tests for tile state and title management hooks.
  * Introduced test for duplicate tile ID detection in layout operations.
  * Enhanced header tests with tile addition workflow validation.
  * Improved sidebar tests for tab navigation and JSON serialization edge cases.

---
**Enterprise sync PR**: https://github.com/voxel51/fiftyone-teams/pull/2858

### CODE REVIEW

**Review by sashankaryal — APPROVED** [review]
nice!

File: app/packages/tiling/src/lib/TilingProvider.test.tsx [hunk 1]
```diff
@@ -262,6 +352,17 @@ describe("TilingProvider", () => {
...
+  describe("useTiling guard", () => {
+    it("throws when called outside a TilingProvider", () => {
+      const consoleError = console.error;
+      console.error = () => {};
+      expect(() => renderHook(() => useTiling())).toThrow(
+        "useTiling must be used inside <TilingProvider>"
+      );
+      console.error = consoleError;
+    });
+  });
```
[the remark this answers is not in the document]
Comment (mozabes) [reply app/packages/tiling/src/lib/TilingProvider.test.tsx:364]: Non-issue
